lib/operations/execute_operation_v2.js

```diff
@@ -0,0 +1,81 @@
+'use strict';
+
+const MongoError = require('mongodb-core').MongoError;
+const Aspect = require('./operation').Aspect;
+
+/**
+ * Executes the given operation with provided arguments.
+ *
+ * This method reduces large amounts of duplication in the entire codebase by providing
+ * a single point for determining whether callbacks or promises should be used. Additionally
+ * it allows for a single point of entry to provide features such as implicit sessions, which
+ * are required by the Driver Sessions specification in the event that a ClientSession is
+ * not provided
+ *
+ * @param {object} topology The topology to execute this operation on
+ * @param {Operation} operation The operation to execute
+ * @param {function} callback The command result callback
+ */
+const executeOperationV2 = (topology, operation, callback) => {
+  if (topology == null) {
+    throw new TypeError('This method requires a valid topology instance');
+  }
+
```

Comment:
Let's make an explicit decision about whether we want to ensure that `operation instanceof OperationBase` or not. The benefit of not explicitly checking is that we can ducktype operations, the benefit of ensuring the check is that we can call `execute` without concern that it doesn't exist on the operation.

Reply:
I think we should ensure that `operation instanceof OperationBase` so I've made that change.

Reply:
Should we not still be checking that `topology != null`?

Reply:
I don't think we get much value out of combining these. I'd prefer splitting into two checks, and throwing informative errors for both cases